af/shovel/aws_s3_lz4cat_sync.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python2
+
+import argparse
+import os
+import shutil
+import tempfile
+from contextlib import contextmanager
+from datetime import datetime
+from subprocess import check_call
+
+if datetime.fromtimestamp(0) != datetime(1970, 1, 1, 0, 0, 0):
+    raise RuntimeError('awscli requires TZ=UTC')
+
+def parse_args():
+    p = argparse.ArgumentParser(description='ooni-pipeline: *.tar.lz4 | tar -I lz4 --extract | aws s3 sync')
+    p.add_argument('--src', metavar='SRC_PATH', help='Source to decompress *.lz4 files from', required=True)
+    p.add_argument('--dst', metavar='DST_PATH', help='Destination root to upload files to, passed to `aws s3 sync`', required=True)
+    opt = p.parse_args()
+    return opt
+
+@contextmanager
+def ScopedTmpdir(*args, **kwargs):
+    tmpdir = tempfile.mkdtemp(*args, **kwargs)
+    try:
+        yield tmpdir
+    finally:
+        shutil.rmtree(tmpdir)
+
+def aws_s3_lz4cat_sync(src, dst):
+    date = os.path.basename(src)
+    for fname in os.listdir(src):
+        if fname == 'index.json.gz': # FIXME: use autoclaving.INDEX_FNAME after moving it to "shared" import with constants
+            continue
+        # Decompressing files chunk-by-chunk saves page cache from pollution.
+        # It's possible to stream files from local lz4 to S3 without touching
+        # the disk and it *seems* to be easy to do that, as 80% of the volume
+        # consists of "large" reports, but it makes usage of `aws s3 sync`
+        # impossible, it would force me to write my own syncing code for
+        # idempotent upload without wasted bandwidth and it's not something I
+        # want to do right now. It may be reconsidered if we want to preserve
+        # YYYY-MM-DD/xxxx naming scheme while transitioning to hourly
+        # batches.  -- @darkk 2017-11-10
```

Comment:
Well... it's slow as hell (time is *hours*):
![airflow-jsonl-n2 2017-11-11 13-23-25](https://user-images.githubusercontent.com/21046/32688543-964beaa4-c6e3-11e7-9bfb-250ba3adfb34.png)

Reply:
That's one bug contributing to slowness: https://github.com/TheTorProject/ooni-pipeline/commit/fd86c9cf45918aef69b8bd49ca436260a09b1827
Another issue is that `aws s3 sync` probably scans whole tree even if there is no `--delete`, so amount of fetched data is ~ O( lz4_files_that_day * report_count_that_day / 2 ). Current solution gives reasonable 15min runtime, but it can probably be further improved.